cpp/harvest/remid_augment.cc

```diff
@@ -20,36 +20,63 @@
 #include <fstream>
 #include <iostream>
 #include <string>
+#include "FileUtil.h"
 #include "MARC.h"
 
 namespace {
 
 [[noreturn]] void Usage() {
-    std::cerr << "Usage: " << ::progname << " marc_input marc_output\n";
-    std::cerr << "       marc_input is the marc input file\n";
-    std::cerr << "       marc_output is the marc output file\n";
+    std::cerr << "Usage: " << ::progname << " <--insert_084_r_v | --serial_splitter> marc_input marc_output [issn_output]\n";
+    std::cerr << "       --insert_084_r_v marc_input marc_output\n";
+    std::cerr << "          marc_input is the marc input file\n";
+    std::cerr << "          marc_output is the marc output file\n";
+    std::cerr << "       --insert_084_r_v marc_input marc_output issn_output\n";
+    std::cerr << "          marc_input is the marc input file\n";
+    std::cerr << "          marc_output is the marc output file without serial\n";
+    std::cerr << "          issn_output is the text output file for issn\n";
     std::exit(EXIT_FAILURE);
 }
 
-} // namespace
+void SerialSplitter(char **argv) {
+    auto input_file(MARC::Reader::Factory(argv[2]));
+    auto output_file_marc(MARC::Writer::Factory(argv[3]));
+    std::set<std::string> issns;
+    bool serial;
+    std::ofstream issn_file(argv[4]);
 
-int Main(int argc, char **argv) {
-    if (argc < 3)
-        Usage();
+    while (MARC::Record record = input_file->read()) {
+        serial = false;
```

Comment:
In these cases you can just move the declaration here as well to save 1 line of code above:
`bool serial(false);`

Reply:
This is still open, even after the changes